### ISSUE TITLE
Javascript API

### ISSUE THREAD

**Issue by tunerooster** [deepch/RTSPtoWebRTC#84]
Your existing implementation (i.e., front-end web page) works great!  For that reason, I would like to be able integrate the same functionality into my own web pages.  Is that possible currently?

That is, is there a way to open/control the same `RTSP` streams created via `config.json` from Javascript in `HTML <video>` elements...  (ie, an API)?

If not, is there any plan to create an API?  Can you summarize what it would take?  That is, would it be a lot of work?

Can it be done using functions in your existing .js code?

Thanks for any help or advice!

**Comment by vdalex25** [comment]
you can look  https://github.com/deepch/RTSPtoWeb  on the fullmultiview page an example of playing multiple streams, including through webrtc

**Comment by tunerooster** [comment]
Dmitriy,

Thanks for your reply!  From what I can tell, the "API" you refer to is for controlling your web page, correct?  What I am after is a Javascript API which would allow me to create my own HTML pages with my own `<video>` elements streaming from your RTSPtoWeb server.

For example, I see a function in your index.js file `makePic(video_element, uuid, chan) ` which appears to attach a stream, originating from your RTSPtoWeb server, to the `src` attribute of your `<video>` element.

I am looking for something like this, but packaged and documented for developer consumption, to attach a stream to *my* src attribute of *my* video element.  Does anything like this exist?  Are there any plans for such a thing?

Or am I completely missing something here?

Please advise... and thanks again for your help!

**Comment by vdalex25** [comment]
are you looking for something like this [WebRtcPlayer](https://github.com/vdalex25/WebRtcPlayer)?

**Comment by tunerooster** [comment]
Yes, that looks very promising.  I'm surprised I did not run it to it in
all my searching.  ...looking forward to trying it out!  Thank you for your
help!

**Comment by vdalex25** [comment]
No wonder you didn't find this before - I only wrote this code yesterday :)

**Comment by tunerooster** [comment]
Wow!  Then double thanks!
